src/overlay/SurveyManager.h

```diff
@@ -86,24 +77,27 @@ class SurveyManager : public std::enable_shared_from_this<SurveyManager>,
 #ifdef BUILD_TESTS
     // Get a reference to the internal `SurveyDataManager` (for testing only)
     SurveyDataManager& getSurveyDataManagerForTesting();
+
+    // Exposes the private function `createTimeSlicedSurveyRequest` for testing
+    // purposes. Sets inbound and outbound peer indicies to 0.
+    std::optional<StellarMessage>
+    createTimeSlicedSurveyRequestForTesting(NodeID const& nodeToSurvey);
 #endif
 
   private:
+    // Stop survey reporting. Must be called after gathering data during the
+    // reporting phase of a survey.
+    void stopSurveyReporting();
     // topology specific methods
     void sendTopologyRequest(NodeID const& nodeToSurvey);
-    void processOldStyleTopologyResponse(NodeID const& surveyedPeerID,
-                                         SurveyResponseBody const& body);
-    void
-    processOldStyleTopologyRequest(SurveyRequestMessage const& request) const;
     void processTimeSlicedTopologyResponse(NodeID const& surveyedPeerID,
-                                           SurveyResponseBody const& body);
+                                           TopologyResponseBodyV2 const& body);
```

Comment:
nit: might be nicer to keep SurveyResponseBody argument to make extensions easier in the future

Reply:
Good point. Done